src/main/scala/cognite/spark/v1/AssetsRelation.scala

```diff
@@ -115,6 +116,43 @@ class AssetsRelation(config: RelationConfig)(val sqlContext: SQLContext)
 
   override def uniqueId(a: AssetsReadSchema): Long = a.id
 
+  implicit val upsertToUpdateTransformer: Transformer[AssetsUpsertSchema, AssetUpdate] =
+    Transformer
+      .define[AssetsUpsertSchema, AssetUpdate]
+      .withFieldComputed(
+        _.labels,
+        u =>
+          u.labels match {
+            case None => None
+            case _ => Some(LabelsOnUpdate(add = Some(u.labels.get.map(CogniteExternalId))))
+        })
+      .buildTransformer
+
+  implicit val createToUpdateTransformer: Transformer[AssetCreate, AssetUpdate] =
```

Comment:
@hakontro Getting some serialization exceptions pointed here, would you have a guess about the reason? :) LabelsOnUpdate encoder is defined within Assets class in scala sdk.

```

Caused by: java.io.NotSerializableException: cognite.spark.v1.AssetsRelation$$anon$4
Serialization stack:
        - object not serializable (class: cognite.spark.v1.AssetsRelation$$anon$4, value: cognite.spark.v1.AssetsRelation$$anon$4@66633170)
        - field (class: cognite.spark.v1.AssetsRelation, name: createToUpdateTransformer, type: interface io.scalaland.chimney.Transformer)
        - object (class cognite.spark.v1.AssetsRelation, cognite.spark.v1.AssetsRelation@154a0cba)
        - element of array (index: 2)
        - array (class [Ljava.lang.Object;, size 3)
        - field (class: java.lang.invoke.SerializedLambda, name: capturedArgs, type: class [Ljava.lang.Object;)

```

Reply:
Hmm maybe I am overriding Setter transformers?

Reply:
I feel like this is about transforming into "Setter". I hoped it would grab Setter Transformer from dataTypes.scala when needed but I think it does not work like that?